banks-server/src/banks_server.rs

```diff
@@ -331,6 +333,58 @@ impl Banks for BanksServer {
             .await
     }
 
+    async fn process_versioned_transaction_with_commitment_and_context(
+        self,
+        _: Context,
+        transaction: VersionedTransaction,
+        commitment: CommitmentLevel,
+    ) -> BanksTransactionResultWithMetadata {
+        let bank = self.bank(commitment);
```

Comment:
I may be dense, but I'm not sure if I understand the approach.  Typically, when processing a transaction with commitment, we simulate the transaction against a bank at a particular commitment, and then send it to the working bank to process and commit the transaction.  With this, we're processing transactions on the older bank, which would break the execution model.

Could you change this to actually processing the transaction on `working_bank` instead, as in the `run()` function?

Reply:
Yeah this was a mistake, thanks for catching this. It should be the working bank.